Java_Problems/RestoreIPAddresses.java

```diff
@@ -6,24 +6,29 @@
 
  Solution: DFS.
  */
-public class Solution {
-    public List<String> restoreIpAddresses(String s) {
-        List<String> res = new ArrayList<String>();
+public class RestoreIPAddresses {
+    public ArrayList<String> restoreIpAddresses(String s) {
+        ArrayList<String> res = new ArrayList<String>();
         dfs(s, new String(), 0, 0, res);
```

Comment:
Our static analysis engine detects that your code is problematic. According to the [Performance (PERFORMANCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#performance-performance), [Dm: Method invokes inefficient new String() constructor (DM_STRING_VOID_CTOR)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dm-method-invokes-inefficient-new-string-constructor-dm-string-void-ctor).

Creating a new `java.lang.String` object using the no-argument constructor wastes memory because the object so created will be functionally indistinguishable from the empty string constant `""`.  Java guarantees that identical string constants will be represented by the same `String` object.  Therefore, you should just use the empty string constant directly.